src/components/dashboard/main-content/course-enrollments/course-cards/UpgradeCourseButton.jsx

```diff
@@ -33,25 +84,25 @@ const UpgradeCourseButton = ({
       'edx.ui.enterprise.learner_portal.course.upgrade_button.clicked',
     );
   };
-
   const handleEnroll = () => {
     sendEnterpriseTrackEvent(
       enterpriseCustomer.uuid,
       'edx.ui.enterprise.learner_portal.course.upgrade_button.to_ecommerce_basket.clicked',
     );
   };
-
   return (
     <>
-      <Button
-        className={className}
-        variant={variant}
-        onClick={handleClick}
-        data-testid="upgrade-course-button"
-      >
-        Upgrade
-        <span className="sr-only">for {title}</span>
-      </Button>
+      <OverlayTriggerWrapper courseRunKey={courseRunKey} hasCourseRunPrice={!!courseRunPrice}>
+        <Button
+          className={className}
+          variant={variant}
+          onClick={handleClick}
+          data-testid="upgrade-course-button"
+        >
+          {intl.formatMessage(messages.upgradeForFreeButton)}
+          <span className="sr-only">for {title}</span>
```

Comment:
[inform] Also ended up tackling this bit of i18n migration in my [draft PR](https://github.com/openedx/frontend-app-learner-portal-enterprise/pull/1118/files#diff-3ef08e773141a24403d51e31c1c6e4f734a206cc43bb38854a19a102fcb0bc7cR100); i think it might make sense to include the `sr-only` `span` as part of the `messages.upgradeForFreeButton` since it's technically all one complete message string, e.g. "Upgrade for free".

[related suggestion] Or,  now that this CTA is renamed to "Upgrade for free" and the word "for" would be repetitive to users of screen readers, what if this was rephrased to "Upgrade {title} for free"?

```jsx
const messages = defineMessages({
    upgradeForFreeButton: {
      id: 'enterprise.learner_portal.dashbboard.enrollments.course.upgrade.button.text',
      defaultMessage: 'Upgrade <s>{title}</s> for free',
      description: 'The label for the course upgrade button text',
  },
});

function getUpgradeButtonScreenReaderText(chunks) {
  return <span className="sr-only">{chunks}</span>;
}

<Button>
  {intl.formatMessage(messages.upgradeForFreeButton, {
    s: getUpgradeButtonScreenReaderText,
    title,
  }}
</Button>
```